### ISSUE TITLE
[packager] Getting sdkman error even though I already set the key and token

### ISSUE THREAD

**Issue by bitspittle** [jreleaser/jreleaser#1064]
My jreleaser config used to work, but since upgrading (from 1.0 to 1.3.1), now it seems to break on sdkman:

```
[ERROR] == JReleaser ==
[ERROR] sdkman.consumerKey must not be blank. Configure a value using the Gradle DSL, or define a JRELEASER_SDKMAN_CONSUMER_KEY environment variable, or define a key/valu
e pair in /home/d9n/.jreleaser/config.properties with key = JRELEASER_SDKMAN_CONSUMER_KEY
[ERROR] sdkman.consumerToken must not be blank. Configure a value using the Gradle DSL, or define a JRELEASER_SDKMAN_CONSUMER_TOKEN environment variable, or define a key/
value pair in /home/d9n/.jreleaser/config.properties with key = JRELEASER_SDKMAN_CONSUMER_TOKEN
```

However, I am setting those, here:

https://github.com/varabyte/kobweb/blob/ee04099f32a1998430c512ea3810048f76e6133b/cli/kobweb/build.gradle.kts#L106

```kotlin
    jreleaser {
        ...
        packagers {
            ...
            val (key, token) = listOf(findProperty("sdkman.key") as? String, findProperty("sdkman.token") as? String)
            if (key != null && token != null) {
                sdkman {
                    consumerKey.set(key)
                    consumerToken.set(token)
                    active.set(Active.RELEASE)
                }
            }
            else {
                println("SDKMAN! packager disabled on this machine since key and/or token are not defined")
            }
        }
```

I also locally confirmed that that block was getting triggered.

I assume there's another place I'm supposed to set the sdkman values now, so I'll look for it, but given that this is a regression, I thought I'd mention it.

**Comment by aalmiray** [comment]
Output of `gradle :cli:kobweb:jreleaserConfig` when no Sdkman key/token are available as env vars, also not available from default secrets file `~/.jreleaser/config.properties`

```
announce:
    enabled: true
    active: always
    sdkman:
        enabled: true
        active: always
        connectTimeout: 20
        readTimeout: 60
        consumerKey: **unset**
        consumerToken: **unset**
        releaseNotesUrl: https://{{repoHost}}/{{repoOwner}}/{{repoName}}/releases/tag/{{tagName}}
        command: major
        extraProperties:
            sdkman__MAGIC_SET__: true
 
packagers:
    sdkman:
        enabled: true
        active: release
        continueOnError: false
        connectTimeout: 20
        readTimeout: 60
        consumerKey: **unset**
        consumerToken: **unset**
 
distributions:
    kobweb:
        enabled: true
        active: always
        type: JAVA_BINARY
        executable:
            name: kobweb
            windowsExtension: bat
        artifacts:
            artifact 0:
                path: /private/tmp/kobweb/cli/kobweb/build/distributions/kobweb-0.9.11.zip
            artifact 1:
                path: /private/tmp/kobweb/cli/kobweb/build/distributions/kobweb-0.9.11.tar
        stereotype: none
        java:
            enabled: true
            version: 11
            groupId: com.varabyte.kobweb.cli
            artifactId: kobweb
            mainClass: MainKt
            multiProject: true
        sdkman:
            enabled: true
            active: release
            continueOnError: false
            candidate: kobweb
            command: major
            releaseNotesUrl: https://{{repoHost}}/{{repoOwner}}/{{repoName}}/releases/tag/{{tagName}}
            connectTimeout: 20
            readTimeout: 60
            consumerKey: **unset**
            consumerToken: **unset**
```

Note that even though key/token are unset model validation still succeeds.

**Comment by aalmiray** [comment]
Output of `gradle :cli:kobweb:jreleaserConfig` using custom login bou din the `build.gradle.kts` file to disable sdkman packager when `sdkman.key` and `sdkman.token` project properties are missing:

```
distributions:
    kobweb:
        enabled: true
        active: always
        type: JAVA_BINARY
        executable:
            name: kobweb
            windowsExtension: bat
        artifacts:
            artifact 0:
                path: /private/tmp/kobweb/cli/kobweb/build/distributions/kobweb-0.9.11.zip
            artifact 1:
                path: /private/tmp/kobweb/cli/kobweb/build/distributions/kobweb-0.9.11.tar
        stereotype: none
        java:
            enabled: true
            version: 11
            groupId: com.varabyte.kobweb.cli
            artifactId: kobweb
            mainClass: MainKt
            multiProject: true
```

**Comment by aalmiray** [comment]
Output of `gradle :cli:kobweb:jreleaserConfig -Psdkman.key=foo -Psdkman.token=bar`

```
announce:
    enabled: true
    active: always
    sdkman:
        enabled: true
        active: always
        connectTimeout: 20
        readTimeout: 60
        consumerKey: ************
        consumerToken: ************
        releaseNotesUrl: https://{{repoHost}}/{{repoOwner}}/{{repoName}}/releases/tag/{{tagName}}
        command: major
        extraProperties:
            sdkman__MAGIC_SET__: true
 
packagers:
    sdkman:
        enabled: true
        active: release
        continueOnError: false
        connectTimeout: 20
        readTimeout: 60
        consumerKey: ************
        consumerToken: ************
 
distributions:
    kobweb:
        enabled: true
        active: always
        type: JAVA_BINARY
        executable:
            name: kobweb
            windowsExtension: bat
        artifacts:
            artifact 0:
                path: /private/tmp/kobweb/cli/kobweb/build/distributions/kobweb-0.9.11.zip
            artifact 1:
                path: /private/tmp/kobweb/cli/kobweb/build/distributions/kobweb-0.9.11.tar
        stereotype: none
        java:
            enabled: true
            version: 11
            groupId: com.varabyte.kobweb.cli
            artifactId: kobweb
            mainClass: MainKt
            multiProject: true
        sdkman:
            enabled: true
            active: release
            continueOnError: false
            candidate: kobweb
            command: major
            releaseNotesUrl: https://{{repoHost}}/{{repoOwner}}/{{repoName}}/releases/tag/{{tagName}}
            connectTimeout: 20
            readTimeout: 60
            consumerKey: ************
            consumerToken: ************
```

Notice that both `consumerKey` and `consumerToken` have their value set.

**Comment by aalmiray** [comment]
The error is reported by the `sdkman` announcer. For historical reasons we've kept this announcer though the preferred way to use sdkman is via the packager as you have it configured. There is indeed a problem between announcer and packager validation.

**Comment by aalmiray** [comment]
Released in v1.4.0 -> https://github.com/jreleaser/jreleaser/releases/tag/v1.4.0
